src/main/scala/org/apache/spark/sql/aggregate/ThetaSketchBuild.scala

```diff
@@ -0,0 +1,137 @@
+import org.apache.spark.sql.catalyst.trees.BinaryLike
+import org.apache.spark.sql.types.{AbstractDataType, DataType, IntegerType, LongType, NumericType, FloatType, DoubleType, ThetaSketchWrapper, ThetaSketchType}
+
+/**
+ * The ThetaSketchBuild function creates a Theta sketch from a column of values
+ * which can be used to estimate distinct count.
+ *
+ * See [[https://datasketches.apache.org/docs/Theta/ThetaSketches.html]] for more information.
+ * 
+ * @param child child expression, from which to build a sketch
+ * @param lgk the size-accraucy trade-off parameter for the sketch
+ */
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = """
+    _FUNC_(expr, lgk) - Creates a Theta Sketch and returns the binary representation.
+      `lgk` (optional, default: 12) the size-accuracy trade-off parameter.""",
+  examples = """
+    Example:
+      > SELECT theta_sketch_get_estimate(_FUNC_(col, 12)) FROM VALUES (1), (2), (3), (4), (5) tab(col);
+       5.0
+  """,
+)
+// scalastyle:on line.size.limit
+case class ThetaSketchBuild(
+    left: Expression,
+    right: Expression,
+    mutableAggBufferOffset: Int = 0,
+    inputAggBufferOffset: Int = 0)
+  extends TypedImperativeAggregate[ThetaSketchWrapper]
+    with BinaryLike[Expression]
+    with ExpectsInputTypes {
+
+  lazy val lgk: Int = {
+    right.eval() match {
+      case null => 12
+      case lgk: Int => lgk
+      case _ => throw new SparkUnsupportedOperationException(
+        s"Unsupported input type ${right.dataType.catalogString}",
+        Map("dataType" -> dataType.toString))
+    }
+  }
+
+  def this(child: Expression) = this(child, Literal(12), 0, 0)
+  def this(child: Expression, lgk: Expression) = this(child, lgk, 0, 0)
+  def this(child: Expression, lgk: Int) = this(child, Literal(lgk), 0, 0)
+
+  override def withNewMutableAggBufferOffset(newMutableAggBufferOffset: Int): ThetaSketchBuild =
+    copy(mutableAggBufferOffset = newMutableAggBufferOffset)
+
+  override def withNewInputAggBufferOffset(newInputAggBufferOffset: Int): ThetaSketchBuild =
+    copy(inputAggBufferOffset = newInputAggBufferOffset)
+
+  override protected def withNewChildrenInternal(newLeft: Expression, newRight: Expression): ThetaSketchBuild = {
+    copy(left = newLeft, right = newRight)
+  }
+
+  override def prettyName: String = "theta_sketch_build"
+
+  override def dataType: DataType = ThetaSketchType
+
+  override def nullable: Boolean = false
+
+  override def inputTypes: Seq[AbstractDataType] = Seq(NumericType, IntegerType, LongType, FloatType, DoubleType)
+
+  override def createAggregationBuffer(): ThetaSketchWrapper = new ThetaSketchWrapper(updateSketch = Some(UpdateSketch.builder().setLogNominalEntries(lgk).build()))
+
+  override def update(wrapper: ThetaSketchWrapper, input: InternalRow): ThetaSketchWrapper = {
+    val value = left.eval(input)
+    if (value != null) {
+      left.dataType match {
```

Comment:
This seems borrowed from KLL and it won't handle things like strings, so that'll certainly need to change